src/core_test/mpas_test_core_string_utils.F

```diff
@@ -0,0 +1,70 @@
+! Copyright (c) 2023, University Corporation for Atmospheric Research (UCAR)
+!
+! Unless noted otherwise source code is licensed under the BSD license.
+! Additional copyright and license information can be found in the 
+! LICENSE file distributed with this code, or at 
+! http://mpas-dev.github.com/license.html .
+!
+module test_core_string_utils
+
+    use mpas_derived_types
+    use mpas_log
+
+    private
+
+    public :: mpas_test_split_string
+
+    contains
+
+    subroutine mpas_test_split_string(err)
+
+        use mpas_string_utils, only : mpas_split_string
+
+        implicit none
+
+        character(len=StrKIND) :: testString
+        character :: delimiter
+        character(len=StrKIND), pointer, dimension(:) :: splitStrings
+        integer, intent(out) :: err
+        integer :: i
+
+        ! Test a basic case
+        delimiter = ' '
+        testString = 'This is a basic test'
+        call mpas_split_string(testString, delimiter, splitStrings)
+        if (trim(splitStrings(1)) /= 'This' .or. &
```

Comment:
If the returned `splitStrings` array contains too few elements, these checks may just result in a segfault. To guard against this possible failure mode, we could first check that the returned `splitStrings` is of the right size.